### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by KaterynaEV**
- [DEMO LINK](https://KaterynaEV.github.io/layout_calendar/)
- [TEST REPORT LINK](https://KaterynaEV.github.io/layout_calendar/report/html_report/)

### CODE REVIEW

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job  👍 
1. The modifier for start day working incorrectly
image

2. Need to implement the modifier for month length

3. This part or the task is incomplete
image

File: src/index.html [hunk 1]
```diff
@@ -7,6 +7,38 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <div class="calendar">
+      <div class="calendar__day calendar__day--start-day_mon"></div>
```
Comment (sTorba24) [inline src/index.html]:
```suggestion
      <div class="calendar__day"></div>
```

File: src/index.html [hunk 2]
```diff
@@ -7,6 +7,38 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <div class="calendar">
```
Comment (sTorba24) [inline src/index.html]:
```suggestion
    <div class="calendar calendar--start-day-mon calendar--month-length-31">
```

Comment (sTorba24) [reply src/index.html]: add modifiers to calendar block

**Review by sTorba24 — CHANGES_REQUESTED** [review]
Almost good! 
still not resolved issue with start day modifier
Maybe you forgot to run deploy?

image

File: src/styles/main.scss [hunk 3]
```diff
@@ -1,3 +1,60 @@
...
+    transform: translateY(-20px);
+  }
+}
+
+.calendar {
+  display: flex;
+  width: $week-width;
+  margin: 20px auto 0;
+  flex-wrap: wrap;
+  flex-direction: row;
```
Comment (andriy-shymkiv) [inline src/styles/main.scss]: it is a default value for flexbox

File: src/styles/main.scss [hunk 4]
```diff
@@ -1,3 +1,60 @@
...
+
 body {
   margin: 0;
 }
+
+@keyframes move {
+  100% {
+    transform: translateY(-20px);
+  }
+}
```
Comment (andriy-shymkiv) [inline src/styles/main.scss]: move it to the end of the file

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1,60 @@
...
+@each $weekday, $column-px in $weekdays {
+  .calendar__day--start-day-#{$weekday} {
+    margin-left: 0;
+  }
+}
+
+@for $day from 28 to 31 {
+  &--month-length-#{$day} &__item:nth-child(n + #{$day + 1}) {
+    display: none;
+    justify-content: center;
```
Comment (andriy-shymkiv) [inline src/styles/main.scss]: do you really need this?
```suggestion
```

**Review by andriy-shymkiv — CHANGES_REQUESTED** [review]

File: src/styles/main.scss [hunk 6]
```diff
@@ -1,3 +1,61 @@
...
+
+.calendar {
+  display: flex;
+  width: $week-width;
+  gap: $gap-size;
+  margin: 20px auto 0;
+  flex-wrap: wrap;
+
+  &__day {
+    box-sizing: border-box;
```
Comment (OleziO) [inline src/styles/main.scss]: Move to the * selector and remove it from all other places.

File: src/styles/main.scss [hunk 7]
```diff
@@ -1,3 +1,61 @@
...
+}
+
+.calendar--start-day-sun .calendar__day:first-child {
+  margin-left: 606px;
+}
+
+@for $i from 1 through 31 {
+  .calendar__day:nth-child(#{$i})::before {
+    content: "#{$i}";
+    font-family: Arial, Helvetica, sans-serif;
```
Comment (OleziO) [inline src/styles/main.scss]: Move to the html selector and remove it from all other places.

**Review by OleziO — CHANGES_REQUESTED** [review]
Fix week-day and month-length selectors
image

**Review by etojeDenys — APPROVED** [review]
good job

**Pull request closed without merging**